iis-wf/iis-wf-export-actionmanager/src/main/java/eu/dnetlib/iis/wf/export/actionmanager/module/CitationsActionBuilderModuleFactory.java

```diff
@@ -23,139 +24,142 @@
-			if (source.getCitations()!=null && source.getCitations().size()>0) {
-				OafEntity.Builder entityBuilder = OafEntity.newBuilder();
-				if (source.getDocumentId()!=null) {
-					entityBuilder.setId(source.getDocumentId().toString());	
-				}
-				ExtraInfo.Builder extraInfoBuilder = ExtraInfo.newBuilder();
-				extraInfoBuilder.setValue(converter.serialize(
-						normalize(source.getCitations())));
-				extraInfoBuilder.setName(EXTRA_INFO_NAME);
-				extraInfoBuilder.setTypology(EXTRA_INFO_TYPOLOGY);
-				extraInfoBuilder.setProvenance(this.inferenceProvenance);
-				extraInfoBuilder.setTrust(getPredefinedTrust());
-				entityBuilder.addExtraInfo(extraInfoBuilder.build());
-				entityBuilder.setType(Type.result);
-				return buildOaf(entityBuilder.build());
-			}
-//			fallback
-			return null;
-		}
-		
-		/**
-		 * Performs confidence level normalization. Removes empty lists.
-		 * Removes 50| prefix from publication identifier.
-		 * @param source
-		 * @return {@link BlobCitationEntry} objects having confidence level value normalized.
-		 */
-		private SortedSet<BlobCitationEntry> normalize(List<CitationEntry> source) {
-			if (source!=null) {
-				SortedSet<BlobCitationEntry> results = new TreeSet<BlobCitationEntry>();
-				for (CitationEntry currentEntry : source) {
-					if (currentEntry.getExternalDestinationDocumentIds().isEmpty()) {
-						currentEntry.setExternalDestinationDocumentIds(null);
-					}
-					if (currentEntry.getDestinationDocumentId()!=null) {
-						currentEntry.setDestinationDocumentId(
-								StringUtils.split(currentEntry.getDestinationDocumentId().toString(), 
-										HBaseConstants.ROW_PREFIX_SEPARATOR)[1]);
-					}
-					results.add(CitationsActionBuilderModuleFactory.build(
-							currentEntry, getConfidenceToTrustLevelNormalizationFactor()));
-				}
-				return results;
-			} else {
-				return null;
-			}
-		}
-	}
+public class CitationsActionBuilderModuleFactory extends AbstractBuilderFactory<Citations> {
+
+    private static final String EXTRA_INFO_NAME = ExtraInfoConstants.NAME_CITATIONS;
+    private static final String EXTRA_INFO_TYPOLOGY = ExtraInfoConstants.TYPOLOGY_CITATIONS;
+
+    // ------------------------ CONSTRUCTORS --------------------------
+    
+    public CitationsActionBuilderModuleFactory() {
+        super(AlgorithmName.document_referencedDocuments);
+    }
+
+    // ------------------------ LOGIC ---------------------------------
+    
+    @Override
+    public ActionBuilderModule<Citations> instantiate(Configuration config, Agent agent, String actionSetId) {
+        return new CitationActionBuilderModule(provideTrustLevelThreshold(config), agent, actionSetId);
+    }
+
+    /**
+     * Creates {@link BlobCitationEntry} from {@link CitationEntry}.
+     * Translates confirence level into trust level applying confidenceToTrustLevelFactor.
+     */
+    public static BlobCitationEntry build(CitationEntry entry, float confidenceToTrustLevelFactor) {
```

Comment:
I guess it was left in factory because it is used in unit tests. This is the case when we change class internals due to external requirements (tests). Eventually we could move it to `CitationsActionBuilderModuleUtils` to make it more straightforward.